src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+        func_800BE504(&this->actor, &this->collider);
+    }
+
+    this->eyeTexIndex = EN_SLIME_EYETEX_OPEN;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->wobbleRot.x = Rand_ZeroOne() * (M_PI * 2.0f);
+    this->wobbleRot.z = Rand_ZeroOne() * (M_PI * 2.0f);
+    ySin = Math_SinS(this->actor.world.rot.y) * 10.0f;
+    yCos = Math_CosS(this->actor.world.rot.y) * 10.0f;
+    effectPos.x = this->actor.world.pos.x + ySin;
+    effectPos.y = this->actor.world.pos.y + 20.0f;
+    effectPos.z = this->actor.world.pos.z + yCos;
+    EffectSsGSplash_Spawn(play, &effectPos, NULL, NULL, 1, 550);
+
+    for (i = 0; i < 10; i++) {
+        effectPos.x = randPlusMinusPoint5Scaled(40.0f) + this->actor.world.pos.x + ySin;
+        effectPos.y = randPlusMinusPoint5Scaled(10.0f) + this->actor.world.pos.y + 40.0f;
+        effectPos.z = randPlusMinusPoint5Scaled(40.0f) + this->actor.world.pos.z + yCos;
+
+        random = Rand_ZeroOne();
+        effectVelocity.x = -Math_SinS(this->actor.world.rot.y) * ((random * 3.5f) + 1.0f);
+        random = Rand_ZeroOne();
+        effectVelocity.z = -Math_CosS(this->actor.world.rot.y) * ((random * 3.5f) + 1.0f);
+        effectVelocity.y = (Rand_ZeroOne() * 6.0f) + 2.0f;
+        EffectSsDtBubble_SpawnCustomColor(play, &effectPos, &effectVelocity, &sBubbleAccel, &sBubblePrimColor,
+                                          &sBubbleEnvColor, Rand_S16Offset(40, 20), 20, 0);
+    }
+
+    if (this->actor.colChkInfo.health == 0) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DEAD);
+    } else {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DAMAGE);
+    }
+    this->actionFunc = EnSlime_UpdateDamageReaction;
+}
+
+/**
+ * Update reaction to taking damage.
+ * Updates the rebuff wobble. Once timer drops to 0, actor initializes death cycle,
+ * re-engages player, or returns to idle.
+ */
+void EnSlime_UpdateDamageReaction(EnSlime* this, PlayState* play) {
+    this->timer--;
+    Math_StepToF(&this->actor.speed, 0.0f, 1.0f);
+    if ((this->timer % 5) == 0) {
+        this->wobbleRot.x = Rand_ZeroOne() * (M_PI * 2.0f);
+        this->wobbleRot.z = Rand_ZeroOne() * (M_PI * 2.0f);
+    }
+    if (this->timer == 0) {
+        if (this->actor.colChkInfo.health != 0) {
+            this->collider.base.acFlags |= AC_ON;
+            if ((this->actor.xzDistToPlayer < 280.0f) && (Player_GetMask(play) != PLAYER_MASK_STONE)) {
+                EnSlime_EngagePlayer(this);
+            } else {
+                EnSlime_ReturnToIdle(this);
+            }
+        } else {
+            SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 40, NA_SE_EN_SLIME_BREAK);
+            EnSlime_InitializeDeath(this);
+        }
+    }
+}
+
+/**
+ * Initialize death action cycle. Disable AC_ON flag,
+ * make actor stop moving, and shrink the actor.
+ * Set action function to Expire.
+ */
+void EnSlime_InitializeDeath(EnSlime* this) {
```

Comment:
Another setup function, we normally call name this one something like `EnSlime_SetupDead` and the one below something like `EnSlime_Dead`.